modules/optimizing-the-vllm-runtime.adoc

```diff
@@ -0,0 +1,169 @@
+endif::[]
+ifndef::upstream[]
+* If you are using specialized {productname-short} groups, you are part of the admin group (for example, `oai-admin-group`) in OpenShift.
+endif::[]
+* To use the vLLM model-serving runtime for speculative decoding with a draft model, you have stored the original model and the speculative model in the same folder within your S3-compatible object storage.
+
+
+.Procedure
+. From the {productname-short} dashboard, click *Settings* > *Serving runtimes*.
++
+The *Serving runtimes* page opens and shows the model-serving runtimes that are already installed and enabled.
++
+. Based on the runtime that you used to deploy your model, perform one of the following actions:
++
+ifdef::upstream[]
+* If you used the pre-installed *vLLM ServingRuntime for KServe* runtime, duplicate the runtime to create a custom version and then follow the remainder of this procedure. For more information about duplicating the pre-installed vLLM runtime, see {odhdocshome}{default-format-url}/serving_models/serving-large-models_serving-large-models#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models[Adding a custom model-serving runtime for the single-model serving platform].
+endif::[]
+ifndef::upstream[]
+* If you used the pre-installed *vLLM ServingRuntime for KServe* runtime, duplicate the runtime to create a custom version and then follow the remainder of this procedure. For more information about duplicating the pre-installed vLLM runtime, see {rhoaidocshome}{default-format-url}/serving_models/serving-large-models_serving-large-models#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models[Adding a custom model-serving runtime for the single-model serving platform].
+endif::[]
+* If you were already using a custom vLLM runtime, click the action menu (&#8942;) next to the runtime and select *Edit*.
++
+The embedded YAML editor opens and shows the contents of the custom model-serving runtime.
+. To configure the vLLM model-serving runtime for speculative decoding by matching n-grams in the prompt, add the following arguments:
++
+[source]
+----
+containers:
+  - args:
+      - --speculative-model=[ngram]
+      - --num-speculative-tokens=[NUM_SPECULATIVE_TOKENS]
+      - --ngram-prompt-lookup-max=[NGRAM_PROMPT_LOOKUP_MAX]
+      - --use-v2-block-manager
+----
++ 
+Replace `[NUM_SPECULATIVE_TOKENS]` and `[NGRAM_PROMPT_LOOKUP_MAX]` with your own values.
+. To configure the vLLM model-serving runtime for speculative decoding with a draft model, do the following:
+.. Remove the `--model` argument:
++
+[source]
+----
+containers:
+  - args:
+      - --model=/mnt/models
+----
+.. Add the following arguments:
++
+[source]
+----
+containers:
+  - args:
+      - --port=8080
+      - --served-model-name={{.Name}}
+      - --distributed-executor-backend=mp
+      - --speculative-model=/mnt/models/<path_to_speculative_model>
```

Comment:
In the above step we removed the model parameter but we have to add add as well  wither in inferec service or in runtime 

```
        - --model=/mnt/models/<path_to_actual_model>
        - --speculative-model=/mnt/models/<path_to_speculative_model>
```